### PR TITLE
Archive activities in the portal  (for WaterScience)

### PR DESCRIPTION
**Pull request by knowuh**
This PR is still in progress.

### CODE REVIEW

File: app/assets/javascripts/components/materials_bin/_fetch_data_mixin.js.coffee [hunk 1]
```diff
@@ -35,5 +35,25 @@ window.MBFetchDataMixin =
           newState[@dataStateKey] = if @processData? then @processData(data) else data
           @setState newState
 
+  archive: (material_id, arhive_url) ->
```
Comment (scytacki) [inline app/assets/javascripts/components/materials_bin/_fetch_data_mixin.js.coffee]: typo in `arhive_url`

File: app/assets/javascripts/components/materials_bin/_fetch_data_mixin.js.coffee [hunk 2]
```diff
@@ -35,5 +35,25 @@ window.MBFetchDataMixin =
...
+    jQuery.ajax
+      url: arhive_url
+      success: (data) =>
+        newState = @state.collectionsData.map (d) ->
+          copy = Object.clone(d)
+          copy.materials = d.materials.filter (m) -> m.id != material_id
+          copy
+        @setState(collectionsData: newState)
+
+  archiveSingle: (material_id, arhive_url) ->
```
Comment (scytacki) [inline app/assets/javascripts/components/materials_bin/_fetch_data_mixin.js.coffee]: typo in `arhive_url`

File: app/controllers/external_activities_controller.rb [hunk 3]
```diff
@@ -288,6 +288,20 @@ def matedit
     end
   end
 
+  def archive
+    authorize @external_activity
+    @external_activity.archive!
+    flash[:notice]= t("matedit.archive_success", {name: @external_activity.name})
+    redirect_to :search  #TBD:  Where to go?
```
Comment (scytacki) [inline app/controllers/external_activities_controller.rb]: I don't know if this was cleared up in the meeting. But I agree that going to search is the best place.

Comment (knowuh) [inline app/controllers/external_activities_controller.rb]: The one problem with this at the moment, is that the flash message doesn't seem to display on the search page.   I am looking into it.

File: spec/models/portal/offering_spec.rb [hunk 4]
```diff
@@ -1,70 +1,69 @@
...
 
     describe "delegates whether the student report" do
       it "is enabled" do
-        @investigation.student_report_enabled = true
-        @offering.student_report_enabled?.should be_true
+        runnable.student_report_enabled = true
+       offering.student_report_enabled?.should be_true
       end
 
       it "is enabled" do
```
Comment (knowuh) [inline spec/models/portal/offering_spec.rb]: Just realized this "it" should be "is disabled"

File: app/views/external_activities/matedit.html.haml [hunk 5]
```diff
@@ -6,8 +6,18 @@
         %span.component_title
         #{@external_activity.name} Editing
     .action_menu_header_right
-      = link_to "Assign To Class", "javascript:void(0)",:onclick=>"get_Assign_To_Class_Popup(#{@external_activity.id}, '#{@external_activity.class.to_s}', null, true)", :class=>"button"
-      = link_to "Edit Resource Options", "javascript:void(0)",:onclick=>"get_edit_resource_popup(#{@external_activity.id}, {use_short_form: true, content_height: 400})", :class=>"button"
+      = link_to t("matedit.archive"),
+        archive_external_activity_path(@external_activity),
+        confirm: t("matedit.archive_confirm"),
+        :class=>"button"
```
Comment (scytacki) [inline app/views/external_activities/matedit.html.haml]: Did Jen and Carolyn like having this be a button? I was thinking a less prominent link would be better. I'm fine with a button if they liked it.

Comment (knowuh) [inline app/views/external_activities/matedit.html.haml]: They liked it.  The didn't love the confirmation box, so I am working on a fix for that.

**Comment by scytacki** [comment]
This looks good to me.

**Comment by knowuh** [comment]
@scytacki I added a few more commits. 

As far as I can tell the failing cuke tests are the usual suspects:

>`cucumber features/teacher_manages_class.feature:54`
>`cucumber features/authors_can_filter_their_own_material.feature:21 `

**Changes in these commits are:**

1. new confirmation dialog box to replace system 'confirm'.
2. fixed appearance of (archive this) text.
3. hid archived activities from a few more views.
4. add a new icon to indicate archived materials in collections management view.

**Comment by knowuh** [comment]
It turns out there is another story to deliver as part of this PR.  Please hold off on the review.

**Comment by scytacki** [comment]
@knowuh This has a conflict now. Probably it is a conflict in schema.rb since I just merged the image library change which changed schema.rb.

**Comment by knowuh** [comment]
@scytacki -- merged master in, keeping your `schema.rb` changes in manual merge.

**Pull request merged**